File: Net5.AspNet.Workshop.Client/ClientApp/src/app/services/data/workshop/workshop-filter.ts
import { Instructor } from "../../../models/instructor";

export class WorkshopFilter {
  constructor() {    
    this.Title = "";
    this.CategoriesIds = new Array<number>();
    this.WorkshopStatesIds = new Array<number>();
    this.InstructorId = 0;

    this.Instructor = new Instructor(); 
  }
  public Title: string;
  public CategoriesIds: Array<number>;
  public WorkshopStatesIds: Array<number>;
  public InstructorId: number;

  public Instructor: Instructor;
  public StartDate?: Date;
  public EndDate?: Date;
}

File: Net5.AspNet.Workshop.Workshop.Api/Infrastructure/FilterParameters/WorkshopFilter.cs
using Net5.AspNet.Workshop.Infrastructure.Dtos;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Net5.AspNet.Workshop.Workshop.Api.Infrastructure.FilterParameters
{
    public class WorkshopFilter
    {
        public WorkshopFilter()
        {
            CategoriesIds = new List<int>();
            WorkshopStatesIds = new List<int>();
        }
        public string Title { get; set; }
        public List<int> CategoriesIds { get; set; }
        public List<int> WorkshopStatesIds { get; set; }
        public int InstructorId { get; set; }

        public InstructorDto Instructor { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
    }
}
